bigchaindb/elections/election.py

```diff
@@ -199,28 +204,38 @@ def has_concluded(self, bigchain, current_votes=[], height=None):
         election_pk = self.to_public_key(self.id)
         votes_committed = self.get_commited_votes(bigchain, election_pk)
         votes_current = self.count_votes(election_pk, current_votes)
-        current_validators = self.get_validators(bigchain, height)
 
-        if self.is_same_topology(current_validators, self.outputs):
-            total_votes = sum(current_validators.values())
-            if (votes_committed < (2/3) * total_votes) and \
-                    (votes_committed + votes_current >= (2/3)*total_votes):
-                return True
+        if self.has_validator_set_changed(bigchain):
+            return False
+
+        current_validators = self.get_validators(bigchain)
+        total_votes = sum(current_validators.values())
+        if (votes_committed < (2/3) * total_votes) and \
+                (votes_committed + votes_current >= (2/3)*total_votes):
+            return True
+
         return False
 
     def get_status(self, bigchain):
         concluded = self.get_election(self.id, bigchain)
         if concluded:
             return self.CONCLUDED
 
+        return self.INCONCLUSIVE if self.has_validator_set_changed(bigchain) else self.ONGOING
+
+    def has_validator_set_changed(self, bigchain):
```

Comment:
Why is this approach preferable over checking the election outputs?

Reply:
It is simpler. I am fine with doing it differently, but we should use the same approach in `get_status` and `has_concluded`.

Reply:
I don't deem either of the approaches simple or difficult but this approach has database calls which can be avoided with the other approach. Since this code will be triggered during `end_block` it would prefer to have the approach with the least amount of database calls.

> we should use the same approach in `get_statu` and `has_concluded`

agreed.

Reply:
Ah, I almost forgot! The output approach does not fit here cause the validator set can be changed and then restored to the original value while we treat elections as inconclusive as soon as the first validator set change happens.